django/db/migrations/state.py

```diff
@@ -187,6 +187,15 @@ def alter_model_options(self, app_label, model_name, options, option_keys=None):
                     model_state.options.pop(key, False)
         self.reload_model(app_label, model_name, delay=True)
 
+    def remove_model_options(self, app_label, model_name, option_name, value_to_remove):
+        model_state = self.models[app_label, model_name]
+        if option_name in model_state.options:
+            objs = model_state.options[option_name]
+            model_state.options[option_name] = [
+                obj for obj in objs if tuple(obj) != tuple(value_to_remove)
+            ]
+        self.reload_model(app_label, model_name, delay=True)
```

Comment:
Can we use `alter_model_options()` with `{"index_together": None}` instead?

Reply:
I think we could use something like:
```
    def remove_model_options(self, app_label, model_name, option_name, value_to_remove):
        model_state = self.models[app_label, model_name]
        if option_name in model_state.options:
            objs = model_state.options[option_name]
            self.alter_model_options(app_label, model_name, {option_name: objs}
```

But I don't think we can entirely drop the `remove_model_options` for `{"index_together": None}`.
We don't necessarily remove _all_ `index_together` values. In theory, we could just rename one of many.
For instance if we go from a model looking like
```
class MyModel(models.Model):
[...]
    class Meta:
        index_together = (
            ("foo", bar"),
            ("bar", baz"),
        )
```
to
```
class MyModel(models.Model):
[...]
    class Meta:
        index_together = (
            ("foo", bar"),
        )
        indexes = [
            models.Index(fields=("bar", baz")),
        ]
```